src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -498,8 +515,22 @@ def makeTransferRucio(self, wflow, dataIn, dids, dataSize, grouping, copies, nod
             # Then make the data subscription, for real!!!
             self.logger.info("Creating rule for workflow %s with %d DIDs in container %s, RSEs: %s, grouping: %s",
                              wflow.getName(), len(dids), dataIn['name'], rseExpr, grouping)
+
+            # define list of rules ids we collect either from createReplicationRule or
+            # moveReplicationRule API calls
+            res = []
             try:
-                res = self.rucio.createReplicationRule(dids, rseExpr, **ruleAttrs)
+                # make decision about current workflow, if it is new request we'll create
+                # new replication rule, otherwise we'll move replication rule
+                if wflow.dataReplacement:
+                    rids = self.getRuleIdsFromDoc(wflow.getName())
+                    self.logger.info("Going to move %s rules for workflow: %s", len(rids), wflow.getName())
+                    for rid in rids:
+                        # the self.rucio.moveReplicationRule may raise different exceptions
+                        # based on different outcome of the operation
+                        res += self.rucio.moveReplicationRule(rid, rseExpr, self.quotaAccount)
```

Comment:
Valentin, as I am certain we will have to eventually debug this, how about making a log record after this move that would say something like:
```
self.logger.info("Rule %s was moved and the new rule is %s", rid, res)
```

Reply:
ok